hera_cal/lst_stack/binning.py

```diff
@@ -565,10 +586,99 @@ def lst_bin_files_for_baselines(
     )
 
     bins = get_lst_bins(lsts, lst_bin_edges)[0]
-    times = np.concatenate(time_arrays)
+    times = np.concatenate([
+        meta.get_transactional("times")[idx] for meta, idx in zip(metas, time_idx)
+    ])
+
     times_in_bins = []
+    lsts_in_bins = []
     for i in range(len(bin_lst)):
         mask = bins == i
         times_in_bins.append(times[mask])
+        lsts_in_bins.append(lsts[mask])
+
+    return bin_lst, data, flags, nsamples, where_inpainted, times_in_bins, lsts_in_bins
+
+
+def lst_bin_files_from_config(
+    config: LSTConfigSingle,
+    bl_chunk_to_load: int | str = 0,
+    nbl_chunks: int = 1,
+    rephase: bool = True,
+    freq_min: float | None = None,
+    freq_max: float | None = None,
+) -> list[UVData] | None:
+
+    if not config.matched_files:
+        # An empty list of files means there's no data to read for this outfile
+        return None
+
+    # get metadata
+    meta = config.config.datameta
+
+    # Split up the baselines into chunks that will be LST-binned together.
+    # This is just to save on RAM.
+    if bl_chunk_to_load == "autos":
+        antpairs = config.autos
+    else:
+        nbls_to_load = int(np.ceil(len(config.antpairs) / nbl_chunks))
+        antpairs = config.antpairs[nbls_to_load * bl_chunk_to_load: nbls_to_load * (bl_chunk_to_load + 1)]
+
+    all_lsts = np.concatenate(config.get_lsts())
+
+    _, data, flags, nsamples, where_inpainted, binned_times, binned_lsts = lst_bin_files_for_baselines(
+        antpairs=antpairs,
+        data_files=config.matched_files,
+        lst_bin_edges=config.lst_grid_edges,
+        freqs=meta.freq_array,
+        pols=config.pols,
+        cal_files=config.calfiles,
+        time_idx=config.time_indices,
+        ignore_flags=False,
+        rephase=rephase,
+        antpos=config.config.reds.antpos,
+        lsts=all_lsts,
+        redundantly_averaged=config.config.is_redundantly_averaged,
+        reds=config.config.reds,
+        freq_min=freq_min,
+        freq_max=freq_max,
+        where_inpainted_files=config.inpaint_files
+    )
+
+    freqs, _ = _get_freqs_chans(meta.freq_array, freq_min, freq_max)
+
+    out = []
+    for (d, f, n, wf, bt) in zip(data, flags, nsamples, where_inpainted, binned_times):
+        # To enable inpaint-mode, set nsamples where things are flagged and inpainted
+        # to zero, and set the flags to false.
+
+        f[wf] = False
+        n[wf] *= -1
+
+        uv = UVData.new(
```

Comment:
As I said elsewhere, I'm skeptical of using UVData objects for bookkeeping like this. It's not really what they were meant for and it doesn't seem to me like we're using much of their capabilities (especially i/o). Seems to me like they are just a way to put the data in a big numpy array that's forcing us to carry around a ton of irrelevant metadata and other baggage. 

Instead, I'd suggest defining our own array format (could match UVData's or could instead let baselines, times, and nights all be separate axes) and then keeping around our "axis labels" i.e. freqs, lsts, nights, bls, as numpy arrays or lists, as appropriate. 

Or maybe I'm misunderstanding how you're using them here...

Reply:
I think we've kinda cleared this up via Slack. But let me say that I think actually UVData objects are about right for this -- the natural shape of the data arrays that come out of the binning are exactly matched to the shape expected by UVData, and most of the metadata we require already exists on the files that were read in. In terms of "extra baggage" -- if you mean extra memory required, I'm not sure this is a big deal here, because the data arrays (and nsamples/flags) are typically much bigger than the metadata in this case. The UVData object is a little wasteful in the metadata since it combines the time/bl axes, so you need to keep around more baselines and times in the metadata than you would otherwise need to, but this shouldn't be a big problem here. 

Furthermore, the new array format I'd have to define would require some kind of smart indexing, which UVData already does for us (the latest UVData does this faster for the rectangular layouts we have here, so I don't think I could make anything faster than already exists). So, in the end, I think this is the right option. The other option would be DataContainers, but this would I think be a non-negligible overhead to construct, and would mean that when doing the averaging over nights later on, that I would have to do this within a loop over baselines (which can get very expensive). 

As a kind of half-way, I've added a new `LSTStack` class which is a very thin wrapper over UVData, and does some of the nice things like splitting the baseline/time axis (only as a view).